packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/TreeNodeRenderer.tsx

```diff
@@ -54,103 +62,136 @@ type TreeNodeRendererProps = Pick<ReturnType<typeof useTree>, "expandNode" | "is
  * @see https://itwinui.bentley.com/docs/tree
  * @public
  */
-export function TreeNodeRenderer({
-  node,
-  expandNode,
-  getIcon,
-  getLabel,
-  onFilterClick,
-  onNodeClick,
-  onNodeKeyDown,
-  isNodeSelected,
-  getHierarchyLevelDetails,
-  reloadTree,
-  actions,
-  ...treeItemProps
-}: TreeNodeRendererProps) {
-  const nodeRef = useRef<HTMLDivElement>(null);
+export const TreeNodeRenderer: ForwardRefExoticComponent<TreeNodeRendererProps & RefAttributes<HTMLElement>> = forwardRef(
+  (
+    {
+      node,
+      expandNode,
+      getIcon,
+      getLabel,
+      onFilterClick,
+      onNodeClick,
+      onNodeKeyDown,
+      isNodeSelected,
+      getHierarchyLevelDetails,
+      reloadTree,
+      actions,
+      ...treeItemProps
+    },
+    forwardedRef,
+  ) => {
+    const nodeRef = useRef<HTMLElement>(null);
+    const ref = useMergedRefs(forwardedRef, nodeRef);
 
-  if (isPlaceholderNode(node)) {
-    return <PlaceholderNode {...treeItemProps} level={node.level} />;
-  }
-
-  if (!isPresentationHierarchyNode(node)) {
-    return <TreeErrorRenderer node={node} getHierarchyLevelDetails={getHierarchyLevelDetails} reloadTree={reloadTree} onFilterClick={onFilterClick} />;
-  }
-
-  const getActions = () => {
-    if (!actions || actions.length === 0) {
-      return undefined;
+    if (isPlaceholderNode(node)) {
+      return <PlaceholderNode {...treeItemProps} level={node.level} ref={ref} />;
     }
 
-    if (actions.length < 4) {
-      return actions.map((action, index) => {
-        const actionInfo = action(node);
-        return <TreeActionButton key={index} {...actionInfo} />;
-      });
+    if (!isPresentationHierarchyNode(node)) {
+      return (
+        <TreeErrorRenderer
+          {...treeItemProps}
```

Comment:
Based on type this does not take everything from `{...treeItemProps}` so only `style` should be passed.

Reply:
changed to only pass properties that exist in TreeErrorRenderer (style).